internal/notification/notification_test.go

```diff
@@ -0,0 +1,63 @@
+package notification
+
+import (
+	"testing"
+	"testing/fstest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestGetTeams(t *testing.T) {
+	testFS := setupTestFS(t)
+
+	tree, err := loadNotifications(testFS)
+	require.NoError(t, err)
+
+	teams := tree.GetTeams()
+	assert.ElementsMatch(t, []string{"test-team"}, teams)
+}
+
+func TestGetProducts(t *testing.T) {
+	testFS := setupTestFS(t)
+
+	tree, err := loadNotifications(testFS)
+	require.NoError(t, err)
+
+	teams := tree.GetProducts("test-team")
+	assert.ElementsMatch(t, []string{"test-product"}, teams)
+}
+
+func TestGetNotification(t *testing.T) {
+	testFS := setupTestFS(t)
+
+	tree, err := loadNotifications(testFS)
```

Comment:
(clarification) If someone drops an invalid notification file in `data/<team>/<notification>.yaml` this should fail the pipeline ?

Reply:
No,this is a fixture created using `testing/fstest` which is an in-memory filesystem. But, if any notifications are invalid yaml this package should panic causing the pipeline to fail. I can actually add a test for that.